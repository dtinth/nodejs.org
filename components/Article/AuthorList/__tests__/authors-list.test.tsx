import React from 'react';
import { render } from '@testing-library/react';
import { LocaleProvider } from '../../../../providers/localeProvider';
import { AppProps } from '../../../../types';
import messages from '../../../../i18n/locales/en.json';
import AuthorsList from '..';

const i18nData = {
  currentLocale: { code: 'en' },
  localeMessages: messages,
} as unknown as AppProps['i18nData'];

describe('AuthorsList component', () => {
  it('renders correctly', () => {
    const authors = ['test-author', 'another-test-author'];
    const { container } = render(
      <LocaleProvider i18nData={i18nData}>
        <AuthorsList authors={authors} />
      </LocaleProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
