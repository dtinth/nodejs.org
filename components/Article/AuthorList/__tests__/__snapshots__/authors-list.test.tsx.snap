// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorsList component renders correctly 1`] = `
<div>
  <div
    class="authorList"
  >
    containers.authorList.title
    <ul>
      <li>
        <a
          aria-label="components.author.githubLinkLabel"
          class="link"
          href="https://github.com/test-author"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span />
        </a>
      </li>
      <li>
        <a
          aria-label="components.author.githubLinkLabel"
          class="link"
          href="https://github.com/another-test-author"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span />
        </a>
      </li>
    </ul>
  </div>
</div>
`;
